trident/ion_balance.py

```diff
@@ -382,25 +412,36 @@ def add_ion_fraction_field(atom, ion, ds, ftype="gas",
     >>> yt.ProjectionPlot(ds, 'x', 'C_p3_ion_fraction').save()
     """
 
-    if particle_type == 'auto':
```

Comment:
Do you think it is worthwhile to have the behavior you added below in a separate function that gets called, since it happens 4 times identically in the code?  I realize that *I* was the person who put the original particle_type check in 4 times, but now it is a bit more bloated and might sit better in a separate function.  If this is too much work, no worries, but it's just a suggestion.